service/resource/s3objectv2/desired.go

```diff
@@ -27,6 +27,10 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 	}
 
 	kmsArn, err := r.awsService.GetKeyArn(clusterID)
+	if IsKeyNotFound(err) {
+		// we can get here during deletion, if he key is already deleted we can safely exit
```

Comment:
Good this is handled and commented. Just a typo on "the key" and can you make it a sentence.

Reply:
Sure, done